entrypoint.sh

```diff
@@ -1,22 +1,41 @@
-#!/bin/sh -l
+#!/bin/bash
 
-echo ""
-echo "Installing dependencies specified in cpanfile"
-echo ""
+set -eo pipefail
 
-cpanm --installdeps .
+# to allow the optional minimum configuration of the CPAN client, as required
+# by some distributions
+if [ "${CPAN_CLIENT}" == 'true' ]
+then
+    echo -e '\nCreating minimal CPAN client configuration\n'
+    echo | cpan
+fi
 
-echo ""
-echo "Executing dzil with the following arguments: "
-echo "---------------------------------------------"
+echo -e '\nInstalling dependencies...\n'
+
+if [ -e ./cpanfile ]
+then
+    cpanm --installdeps .
+else
+    dzil authordeps --missing | cpanm --notest
+    dzil listdeps --missing | cpanm --notest
+fi
+
+set -u
+
+LINE='---------------------------------------------'
+
+echo -e '\nExecuting dzil with the following arguments:\n'
+echo $LINE
 echo "$@"
-echo "---------------------------------------------"
-echo ""
+echo -e "${LINE}\n"
+
+set +e
 
-$@
+dzil $@
```

Comment:
I wonder if I was expecting the complete line?

I might have used this for debugging, anyway it looks better and I will update the documentation on this.

Reply:
That's a good one: the Ebirah image uses `dzil` as `ENTRYPOINT`, but the `Dockerfile` in this repository overwrites it and does not invoke `dzil` again.
During my local tests (outside the Github environment) that didn't work until I added the line 34.